src/metatensor/models/experimental/alchemical_model/tests/test_regression.py

```diff
@@ -21,16 +21,14 @@
 from . import DATASET_PATH
 
 
-torch.set_default_dtype(torch.float64)
-
-
 def test_regression_init():
     """Perform a regression test on the model at initialization"""
 
     # reproducibility
     random.seed(0)
     np.random.seed(0)
     torch.manual_seed(0)
+    torch.set_default_dtype(torch.float64)
```

Comment:
Can you reset it to float32 at the end of the test? I think this is necessary in order for it not to carry over (although I'm not 100% sure)